Enemy Waves

Migu3lito5 opened the pull request:
Added Waves to the game loop. 

At first 3 enemy missiles spawn with 0 enemies, once a timer runs out, the next wave starts, which increases the enemy amount by one. So for example, enemy missiles goes to 4 and a enemy goes to 1.

Migu3lito5 committed:
Updating Scene for wave feature

Migu3lito5 committed:
Updated enemy and missile scripts for waves to work

Migu3lito5 committed:
Update MiguelScene.unity

ConlynPattison commented on Seasons Defense/Assets/Scripts/EnemySpawner.cs:
```diff
@@ -26,19 +31,32 @@ private IEnumerator SpawnEnemies(float interval, GameObject[] enemy)
+
+            enemyCount = enemyAtStart + 1;
+            yield return new WaitForSeconds(9);
+            LevelManager.Instance.finishedSpawningEnemies = true;
```
Should this line go outside of the outer while loop? Since the spawning of enemies could go on for another wave after this assignment

ConlynPattison committed:
Minor adjustments to code.
Moved the modification of the spawning boolean outside of the while loops
and refactored multiple operations and names.

ConlynPattison reviewed (approved):
Made some minor changes myself, looks good!
